charts/hivemq-edge/templates/statefulset.yml

```
@@ -168,7 +168,69 @@ spec:
                   name: "{{ include "hivemq-edge.name" (dict "prefix" "hivemq-edge" "name" "clientauth" "releaseName" .Release.Name) }}"
                   key: truststore-password
             {{- end }}
-            {{- if .Values.admin.secret.enabled }}
+            {{- if and (.Values.adminLdap.tls.enabled) (ne .Values.adminLdap.tls.create.enabled true) }}
+            - name: HIVEMQ_LDAP_TRUSTSTORE_PATH
+              value: /ldap-trust/{{ .Values.adminLdap.tls.truststore.truststoreSecretKey }}
+            - name: HIVEMQ_LDAP_TRUSTSTORE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.adminLdap.tls.truststore.passwordSecretName }}"
+                  key: "{{ .Values.adminLdap.tls.truststore.passwordSecretKey }}"
+            {{- end }}
+            {{- if and (.Values.adminLdap.tls.enabled) (.Values.adminLdap.tls.create.enabled) }}
+            - name: HIVEMQ_LDAP_TRUSTSTORE_PATH
+              value: /ldap-trust/truststore.jks
+            - name: HIVEMQ_LDAP_TRUSTSTORE_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ include "hivemq-edge.name" (dict "prefix" "hivemq-edge" "name" "ldap" "releaseName" .Release.Name) }}"
+                  key: truststore-password
+            {{- end }}
+            {{- if .Values.adminLdap.enabled }}
+            - name: HIVEMQ_LDAP_ENABLED
+              value: "true"
+            - name: HIVEMQ_LDAP_TLS_MODE
+              value: "{{ .Values.adminLdap.tlsMode }}"
+            - name: HIVEMQ_LDAP_SIMPLEBIND_RDNS
+              value: "{{ .Values.adminLdap.simplebind.userDn }}"
+            - name: HIVEMQ_LDAP_SIMPLEBIND_PASSWORD
+              value: "{{ .Values.adminLdap.simplebind.password }}"
+            - name: HIVEMQ_LDAP_UID
+              value: "{{ .Values.adminLdap.uid }}"
+            - name: HIVEMQ_LDAP_RDNS
+              value: "{{ .Values.adminLdap.rdns }}"
+            - name: HIVEMQ_LDAP_DIRECTORY_DESCENT
+              value: "{{ .Values.adminLdap.directoryDescent }}"
+            - name: HIVEMQ_LDAP_MAX_CONNECTION
+              value: "{{ .Values.adminLdap.maxConnections }}"
+            - name: HIVEMQ_LDAP_CONNECT_TIMEOUT_MS
+              value: "{{ .Values.adminLdap.connectionTimeoutMillis }}"
+            - name: HIVEMQ_LDAP_SEARCH_TIMEOUT_S
+              value: "{{ .Values.adminLdap.searchTimeoutSeconds }}"
+              {{- if and (hasKey .Values.adminLdap "objectClass") .Values.adminLdap.objectClass }}
```

The `hasKey` here is redundant. If the value doesn't exist, it will return empty.

```suggestion
              {{- if .Values.adminLdap.objectClass }}
```